static/js/services/delete-doc.js

```diff
@@ -1,66 +1,65 @@
+var _ = require('underscore');
+
 (function () {
 
   'use strict';
 
   var inboxServices = angular.module('inboxServices');
 
-  inboxServices.factory('DeleteDoc', ['DB',
-    function(DB) {
+  inboxServices.factory('DeleteDoc',
+    function(
+      $q,
+      DB
+    ) {
+
+      'ngInject';
 
-      var updateParent = function(doc, callback) {
+      var getParent = function(doc) {
         if (doc.type === 'person' && doc.parent && doc.parent._id) {
-          DB.get()
+          return DB.get()
             .get(doc.parent._id)
             .then(function(parent) {
               if (parent.contact.phone !== doc.phone) {
-                return callback();
+                return;
               }
               parent.contact = null;
-              DB.get()
-                .put(parent)
-                .then(function() {
-                  callback();
-                })
-                .catch(function(err) {
-                  callback(err);
-                });
-            })
-            .catch(function(err) {
-              if (err.reason === 'deleted') {
-                return callback();
-              }
-              return callback(err);
+              return parent;
             });
-        } else {
-          callback();
         }
+        return $q.resolve();
       };
 
-      return function(docId, callback) {
-        DB.get()
-          .get(docId)
-          .then(function(doc) {
-            updateParent(doc, function(err) {
-              if (err) {
-                return callback(err);
+      /**
+       * Delete the given docs. If 'person' type then also fix the
+       * contact hierarchy.
+       *
+       * @param docs {Object|Array} Document or array of documents to delete.
+       */
+      return function(docs) {
+        if (!_.isArray(docs)) {
+          docs = [ docs ];
+        }
+        var toUpdate = docs.map(function(doc) {
+          return {
+            _id: doc._id,
+            _rev: doc._rev,
+            _deleted: true
+          };
+        });
+        return $q.all(docs.map(function(doc) {
```

Comment:
Hmm what if you are running `DeleteDocs([doc, doc.parent])`? 
Then `doc.parent` is being edited twice in parallel, right? Is it going to be consistent, been the get and the put? Are we saved by the single-threadedness?